renderer/components/action-bar/controls/advanced.js

```diff
@@ -0,0 +1,204 @@
+// Packages
+import PropTypes from 'prop-types';
+import React from 'react';
+import css from 'styled-jsx/css';
+
+// Vectors
+import {
+  SwapIcon,
+  BackIcon,
+  LinkIcon,
+  DropdownArrowIcon
+} from '../../../vectors';
+
+// Containers
+import {connect, ActionBarContainer, CropperContainer} from '../../../containers';
+
+// Utilities
+import {handleWidthInput, handleHeightInput} from '../../../utils/inputs';
+
+const advancedStyles = css`
+  .advanced {
+    height: 50px;
+    display: flex;
+    flex: 1;
+    align-items: center;
+    justify-content: space-between;
+  }
+`;
+
+const AdvancedControls = {};
+
+class Left extends React.Component {
+  render() {
+    const {toggleAdvanced, toggleRatioLock, ratioLocked, ratio = []} = this.props;
+
+    return (
+      <div className="advanced">
+        <BackIcon onClick={toggleAdvanced}/>
+        <div className="select">
+          <span>{ratio[0]}:{ratio[1]}</span>
+          <DropdownArrowIcon size={15}/>
+        </div>
+        <LinkIcon active={ratioLocked} onClick={() => toggleRatioLock()}/>
+        <style jsx>{advancedStyles}</style>
+        <style jsx>{`
+          .select {
+            border: 1px solid #ddd;
+            border-radius: 4px;
+            font-size: 0.7rem;
+            height: 1.1rem;
+            transition: border 0.12s ease-in-out;
+            display: flex;
+            align-items: center;
+            padding: 0 5px;
+          }
+
+          .select span {
+            flex: 1;
+            text-align: right;
+            padding-right: 5px;
+          }
+
+          .select:hover {
+            border-color: #ccc;
+          }
+        `}</style>
+      </div>
+    );
+  }
+}
+
+Left.propTypes = {
+  toggleAdvanced: PropTypes.func.isRequired,
+  toggleRatioLock: PropTypes.func.isRequired,
+  ratioLocked: PropTypes.bool,
+  ratio: PropTypes.array
+};
+
+AdvancedControls.Left = connect(
+  [ActionBarContainer, CropperContainer],
+  ({ratioLocked}, {ratio}) => ({ratio, ratioLocked}),
+  ({toggleAdvanced, toggleRatioLock}) => ({toggleAdvanced, toggleRatioLock})
+)(Left);
+
+class Right extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      width: props.width,
+      height: props.height
+    };
+  }
+
+  componentWillReceiveProps(newProps) {
```

Comment:
This method is deprecated: https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops